Makefile

```diff
@@ -1,18 +1,27 @@
+.PHONY: all phony container push clean node-problem-detector
+
 all: push
 
 # See node-problem-detector.yaml for the version currently running-- bump this ahead before rebuilding!
 TAG = v0.2
 
 PROJ = google_containers
 
-node-problem-detector: node_problem_detector.go
+PKG_SOURCES := $(shell find pkg -name '*.go')
```

Comment:
It's a premature optimization. Managing the timestamps for that many files is a bit slower and having to do `make clean` after a vendor update doesn't seem like that big of a deal.